app/models/article.rb

```diff
@@ -1,4 +1,25 @@
 class Article < ApplicationRecord
     validates :title, presence: true, uniqueness: true
     validates :body, presence: true
+    validates :slug, uniqueness: true
+    
+    after_create :update_slug
+    before_update :assign_slug
+
+    def update_slug
+        update_attribute(:slug, assign_slug)
+    end
+
+    def create_slug
+        return self.title.parameterize
+    end
+
+    def to_param
+       slug
+    end
+
+    private
+    def assign_slug
+        self.slug = create_slug
+    end

```

Comment:
It probably doesn’t matter for a project of this size, but I’m noticing a code smell. You have a bunch of methods in this class that are related not to the article it models, but to the slug. That is usually an indication that you should create a separate slug class.